matchtree_test.go

```diff
@@ -306,3 +306,26 @@ func TestBranchesRepos(t *testing.T) {
 		t.Fatalf("expect %d documents, but got at least 1 more", len(want))
 	}
 }
+
+func TestRepoIds(t *testing.T) {
+	d := &indexData{
+		repoMetaData:    []Repository{{Name: "r0", ID: 0}, {Name: "r1", ID: 1}, {Name: "r2", ID: 2}, {Name: "r3", ID: 3}},
+		fileBranchMasks: []uint64{1, 1, 1, 1, 1, 1},
+		repos:           []uint16{0, 0, 1, 2, 3, 3},
+	}
+	mt, err := d.newMatchTree(&query.RepoIds{Repos: roaring.BitmapOf(1, 3, 99)})
+	if err != nil {
+		t.Fatal(err)
+	}
+	want := []uint32{2, 4, 5}
```

Comment:
This test is pretty much the same as the `TestRepoSet` above but I did find it quite difficult to wrap my head around what the `want := []uint32{2, 4, 5}` was supposed to mean here. I worked out eventually that they are the indexes in the `repos` array above that correspond to the repos we're filtering for. I probably don't want to refactor all these tests in this MR but it would be nice if we could think of a way to make it more obvious the relationship between the `repos` above and where these `2,4,5` come from.